lib/table.c

```diff
@@ -790,6 +796,12 @@ mutation_table_append_columns(mutation_table_t *self, size_t num_rows, site_id_t
             num_rows * sizeof(node_id_t));
     memcpy(self->derived_state + self->total_derived_state_length, derived_state,
             total_derived_state_length * sizeof(char));
+    if (parent == NULL) {
+        /* If parent is NULL, set all parents to the null mutation */
+        memset(self->parent + self->num_rows, 0xff, num_rows * sizeof(mutation_id_t));
```

Comment:
Why is this the null mutation?

Reply:
Why do we set the parent to the null mutation by default, or why is 0xff the null mutation?

Reply:
Sorry, the second -- why 0xff?

Reply:
The null mutation/node is -1, which is all 1s at the bit level. 0xf is hex for all 1s. Memset takes a byte as it's argument, so you have to specify 0xff to fill a buffer with 1 bits. It's not that obvious, and is a departure from using NULL_MUTATION where possible but it is convenient and fast.